src/dispatch/signal/views.py

```diff
@@ -73,47 +74,40 @@ def create_signal_instance(
     )
 
     if not signal_instance_in.signal:
+        # we try to get the signal definition by external id or variant
         external_id = signal_instance_in.raw.get("externalId")
         variant = signal_instance_in.raw.get("variant")
+        signal_definition = signal_service.get_by_variant_or_external_id(
+            db_session=db_session,
+            project_id=project.id,
+            external_id=external_id,
+            variant=variant,
+        )
 
-        if external_id or variant:
-            signal = signal_service.get_by_variant_or_external_id(
-                db_session=db_session,
-                project_id=project.id,
-                external_id=external_id,
-                variant=variant,
-            )
-
-            signal_instance_in.signal = signal
-        else:
-            msg = "An external id or variant must be provided."
-            log.warn(msg)
-            raise HTTPException(
-                status_code=status.HTTP_400_BAD_REQUEST,
-                detail=[{"msg": msg}],
-            ) from None
+    if not signal_definition:
+        # we get the default signal definition
+        signal_definition = signal_service.get_default(
+            db_session=db_session,
+            project_id=project.id,
+        )
+        msg = "Default signal definition used for signal instance with external id {external_id} or variant {variant}."
+        log.warn(msg)
 
-    if not signal:
-        msg = f"No signal definition found. External Id: {external_id} Variant: {variant}"
+    if not signal_definition:
+        msg = f"No signal definition could be found by external id {external_id} or variant {variant}, and no default exists."
         log.warn(msg)
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail=[{"msg": msg}],
         ) from None
 
-    if not signal.enabled:
-        msg = f"Signal definition not enabled. Signal Name: {signal.name}"
-        log.info(msg)
-        raise HTTPException(
-            status_code=status.HTTP_403_FORBIDDEN,
-            detail=[{"msg": msg}],
-        ) from None
+    signal_instance_in.signal = signal_definition
 
     try:
         signal_instance = signal_service.create_instance(
             db_session=db_session, signal_instance_in=signal_instance_in
         )
-        signal_instance.signal = signal
+        signal_instance.signal = signal_definition
```

Comment:
I'm not sure this is needed if we are already doing the association [here](https://github.com/Netflix/dispatch/blob/master/src/dispatch/signal/service.py#L539)

Reply:
Yeah, seems redundant -- also in `create_instance` we refetch it and I am not sure that's necessary either.

```python
signal = get(db_session=db_session, signal_id=signal_instance_in.signal.id)
```

Reply:
Yeah, we're already including signal in `signal_instance_in`. Also, the `db_session.commit()` [here](https://github.com/Netflix/dispatch/blob/master/src/dispatch/signal/views.py#L117) seems redundant too as we already do it at the end of `create_instance`

Reply:
The get inside `create_instance` was needed. Otherwise, the app threw a `AttributeError: 'SignalRead' object has no attribute '_sa_instance_state'` exception